cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/DatasetInputFormatProvider.java

```diff
@@ -62,6 +65,9 @@ public String getInputFormatClassName() {
   @Override
   public Map<String, String> getInputFormatConfiguration() {
     if (dataset instanceof InputFormatProvider) {
+      if (splits != null) {
+        LOG.warn("Dataset of type InputFormatProvider does not support splits");
```

Comment:
How about "Ignoring user-specified splits for dataset %s because it is an InputFormatProvider"?